src/cli/config.js

```diff
@@ -14,139 +14,36 @@ import {
-    excludes: Array<RegExp>,
-    options: ExternalOptions,
-    extends?: string,
-};
-
-const subSchema = {...configSchema, required: []};
-subSchema.properties = {...configSchema.properties, extends: {type: 'string'}};
-delete subSchema.properties.schemaFilePath;
-
-export const loadSubConfigFile = (configFile: string): SubConfig => {
-    const jsonData = fs.readFileSync(configFile, 'utf8');
+    const data: Config = JSON.parse(fs.readFileSync(configFile, 'utf8'));
     // eslint-disable-next-line flowtype-errors/uncovered
-    const data: JSONSubConfig = JSON.parse(jsonData);
-    const toplevelKeys = ['excludes', 'options', 'extends'];
-    Object.keys(data).forEach((k) => {
-        if (!toplevelKeys.includes(k)) {
-            throw new Error(
-                `Invalid attribute in non-root config file ${configFile}: ${k}. Allowed attributes: ${toplevelKeys.join(
-                    ', ',
-                )}`,
-            );
-        }
-    });
-    validateOrThrow(data, subSchema);
-    return {
-        excludes: data.excludes?.map((string) => new RegExp(string)) ?? [],
-        options: data.options ?? {},
-        extends: data.extends ?? '',
-    };
-};
-
-export const loadDirConfigFiles = (
-    filesResponse: string,
-    rootConfig: {path: string, config: CliConfig},
-): {[dir: string]: SubConfig} => {
-    const dirConfigMap: {[key: string]: SubConfig} = {};
-
-    // TODO: circular extends will cause infinite loop... consider instrumenting code to monitor for loops in the future?
-    const loadExtendedConfig = (configPath: string): SubConfig => {
-        let dirConfig = loadSubConfigFile(configPath);
-        if (dirConfig.extends) {
-            const isRootConfig = dirConfig.extends === rootConfig.path;
-            const {options, excludes} = isRootConfig
-                ? rootConfig.config
-                : addConfig(dirConfig.extends);
-            dirConfig = extendConfig({options, excludes}, dirConfig);
-        }
-        return dirConfig;
-    };
-    const addConfig = (configPath) => {
-        const {dir} = path.parse(configPath);
-        if (dirConfigMap[dir]) {
-            return dirConfigMap[dir];
-        }
-        dirConfigMap[dir] = loadExtendedConfig(configPath);
-        return dirConfigMap[dir];
-    };
-    const extendConfig = (
-        toExtend: SubConfig,
-        current: SubConfig,
-    ): SubConfig => ({
-        // $FlowFixMe[exponential-spread]
-        options: {...toExtend.options, ...current.options},
-        excludes: Array.from(
-            new Set([...toExtend.excludes, ...current.excludes]),
-        ),
-    });
-
-    filesResponse
-        .trim()
-        .split('\n')
-        .forEach((configPath) => {
-            const {dir} = path.parse(configPath);
-            if (dir && !dirConfigMap[dir]) {
-                dirConfigMap[dir] = loadExtendedConfig(configPath);
-            }
-        });
-    return dirConfigMap;
+    validateOrThrow(data, configSchema);
+    return data;
```

Comment:
Looks like you deleted the logic that protects this return from being `null`/`undefined`, and also the mappings for `excludes` and `schemaFilePath`. If that was intended, then LGTM.